lib/elixir/lib/exception.ex

```diff
@@ -1150,9 +1150,21 @@ defmodule Protocol.UndefinedError do
 
   @impl true
   def message(%{protocol: protocol, value: value, description: description}) do
-    "protocol #{inspect(protocol)} not implemented for #{inspect(value)}" <>
-      maybe_description(description) <> maybe_available(protocol)
-  end
+    "protocol #{inspect(protocol)} not implemented for #{inspect(value)} of type " <>
+      value_type(value) <> maybe_description(description) <> maybe_available(protocol)
+  end
+
+  defp value_type(value) when is_bitstring(value), do: "BitString"
```

Comment:
@treble37 this looks excellent, we are just missing the `__struct__` type. Something like:

    defp value_type(%{__struct__: struct}) when is_atom(struct), do: inspect(struct)

Can you please add this clause and add a test accordingly if we don't have any? Thank you.

Reply:
Will do, thank you for the feedback!

Reply:
Updated @josevalim 😃